node/src/reactor/main_reactor/keep_up.rs

```diff
@@ -688,9 +687,22 @@ impl MainReactor {
                     Err(err) => Err(err.to_string()),
                 }
             }
-            None => {
-                debug!("historical: did not find any orphaned block headers");
-                Ok(None)
+            HighestOrphanedBlockResult::MissingFromBlockHeightIndex(block_height) => {
+                if block_height == 0 {
+                    Ok(Some(SyncBackInstruction::GenesisSynced))
```

Comment:
yes, that's true. the real issue is it should not be possible for completed_blocks to have _any_ entry that doesn't have a corresponding entry in the block height index and / or that isn't actually complete. 

thus, this should be unreachable. but if it were reached, for the purpose of this logic (historical state) which is concerned with deciding whether the node should attempt to get the parent of this block (which will fail if attempted if this block is not actually complete), it seemed preferable to allow the node to keep running forward if the low bound is reported as 0 since we know that the node wont attempt to sync its parent (a non existent block at block_height == -1) and thus naturally avoids the pitfall. ie since height 0 is the left bound terminator, the process will not run itself off the edge if allowed to continue.

but, i will remove this special case and have it Err even if height is 0.